ksql-serde/src/main/java/io/confluent/ksql/serde/json/KsqlJsonDeserializer.java

```diff
@@ -131,29 +118,32 @@ private Object enforceFieldType(final Schema fieldSchema, final Object columnVal
   }
 
   private List<?> enforceFieldTypeForArray(final Schema fieldSchema, final List<?> arrayList) {
-    return arrayList.stream()
-        .map(item -> enforceFieldType(fieldSchema.valueSchema(), item))
-        .collect(Collectors.toList());
+    final List<Object> array = new ArrayList<>(arrayList.size());
+    for (final Object item : arrayList) {
+      array.add(enforceFieldType(fieldSchema.valueSchema(), item));
+    }
+    return array;
   }
 
   private Map<String, Object> enforceFieldTypeForMap(
       final Schema fieldSchema,
-      final Map<String, Object> columnMap) {
-    return columnMap.entrySet().stream()
-        .collect(Collectors.toMap(
-            e -> enforceFieldType(Schema.OPTIONAL_STRING_SCHEMA, e.getKey()).toString(),
-            e -> enforceFieldType(fieldSchema.valueSchema(), e.getValue())
-        ));
+      final Map<String, ?> columnMap) {
+    final Map<String, Object> ksqlMap = new HashMap<>();
+    for (final Map.Entry<String, ?> e : columnMap.entrySet()) {
+      ksqlMap.put(
+          enforceFieldType(Schema.OPTIONAL_STRING_SCHEMA, e.getKey()).toString(),
+          enforceFieldType(fieldSchema.valueSchema(), e.getValue())
+      );
+    }
+    return ksqlMap;
   }
 
   private Struct enforceFieldTypeForStruct(
       final Schema fieldSchema,
       final Map<String, Object> structMap) {
     final Struct columnStruct = new Struct(fieldSchema);
-    final Map<String, String> caseInsensitiveFieldNameMap =
-        getCaseInsensitiveFieldNameMap(fieldSchema.fields());
     final Map<String, String> caseInsensitiveStructFieldNameMap =
-        getCaseInsensitiveStructFieldNameMap(structMap);
+        getCaseInsensitiveFieldNameMap(structMap, false);
```

Comment:
If `omitAt` is always going to be `false`, do we need it?

Reply:
Its true on line 82